### DISCUSSION TITLE
8bit parralel ili9341 driver for micropython?

### DISCUSSION THREAD

**Discussion by skeyvin** [micropython/micropython#12043] [RP2040 / Pico] [answered]
hi i have an uno 2.4inch tft shield that uses 8bit parallel bus for data transfer i can use mcufriends_kvb to use it with arduino, is there a library for the 8bit parallel  ili9341 for me to use?

**Comment by bidrohini** [comment]
Yes, there are libraries available for using the ILI9341 TFT display with an 8-bit parallel interface on Arduino. One popular library that supports the ILI9341 with an 8-bit parallel bus is the "Adafruit_ILI9341" library by Adafruit.

To use the "Adafruit_ILI9341" library, follow these steps:

1. Install the Library:
   Open the Arduino IDE, go to "Sketch" > "Include Library" > "Manage Libraries...". In the Library Manager, search for "Adafruit ILI9341" and click the "Install" button to install the library.

2. Wiring:
   Connect your ILI9341 TFT display to your Arduino board using an 8-bit parallel connection. Make sure you wire the data pins (D0-D7), the control pins (CS, DC, RESET), and the backlight control pin (if applicable) correctly.

3. Initialize the Library:
   In your Arduino sketch, include the "Adafruit_ILI9341" library at the beginning of your code:

#include <Adafruit_ILI9341.h>


4. Create an Instance of the Display:
   Create an instance of the ILI9341 display by declaring an object of the `Adafruit_ILI9341` class. You need to provide the appropriate pin numbers for CS, DC, and RESET as parameters. For example:


#define TFT_CS    10
#define TFT_DC    9
#define TFT_RESET 8

Adafruit_ILI9341 tft = Adafruit_ILI9341(TFT_CS, TFT_DC, TFT_RESET);

5. Initialize the Display:
   In your `setup()` function, initialize the display with the appropriate rotation and call the `begin()` function:


void setup() {
  tft.begin();
  tft.setRotation(3); // Adjust the rotation if needed (0, 1, 2, or 3)
  // Other setup code...
}


6. Display Graphics:
   You can now use the various functions provided by the library to display graphics, text, shapes, etc. on the ILI9341 TFT display.

Please note that the pin numbers used in the example above (TFT_CS, TFT_DC, TFT_RESET) are just placeholders. You should replace them with the actual pin numbers you are using to connect the ILI9341 display to your Arduino.

The "Adafruit_ILI9341" library provides a wide range of features and functions to interact with the display, including drawing shapes, displaying text, and using graphics acceleration features (if supported by the display). You can refer to the Adafruit documentation and examples for more detailed usage instructions and examples.

Always double-check your wiring and connections to ensure they are correct before running any code to avoid potential hardware issues.

**Reply by scruss** [reply]
please don't paste chatgpt answers here. What you pasted is utterly irrelevant to the OP's question.

**Comment by robert-hh** [comment, marked as answer]
I's unclear if you want to use it with C code or Python. If you want to use Python, you can have a look at the Pico variant of the SSD1963 driver here, which also uses an 8 bit 8080-type parallel bus. There are PIO scripts to drive the bus, especially file https://github.com/robert-hh/SSD1963-TFT-Library-for-PyBoard-and-RP2040/blob/master/rp2040/tft_pio.py. Granted, it may be hard to find the right parts in the file to use. The actual PIO state machine are at the beginning of the class and are usually only a few bytes long.
